dsl/src/lib.rs

```diff
@@ -71,17 +92,29 @@ fn parse_literal(it: &mut Peekable<IntoIter>) -> ParsedExpr {
     }
 
     // integer literals
-    if let Ok(Literal::Integer(i)) = Literal::try_from(token) {
-        let v = i.value::<u64>().expect("i don't think this integer fits?");
-        return ParsedExpr {
-            t: ExprType::Int,
-            tokens: quote! { #v },
-        };
-    }
+    match Literal::try_from(token) {
+        Ok(Literal::Integer(i)) => {
+            let v = i.value::<u64>().expect("i don't think this integer fits?");
+            return ParsedExpr {
+                t: ExprType::Int,
+                tokens: quote! { #v },
+            };
+        },
+
+        Ok(Literal::String(lit)) => {
+            let v = lit.value();
+            return ParsedExpr {
+                t: ExprType::Domain,
+                tokens: quote!{ Domain::new(#v.parse()?) },
+            };
+        },
 
-    ParsedExpr {
-        t: ExprType::Nil,
-        tokens: TokenStream::new(),
+        _ => {
```

Comment:
Some diagnostics here. You can bind it to 

```suggestion
        val => {
            println!("{val:?}");
```